hub/core/storage/local.py

```diff
@@ -94,7 +96,9 @@ def __delitem__(self, path: str):
             KeyError: If an object is not found at the path.
             DirectoryAtPathException: If a directory is found at the path.
             Exception: Any other exception encountered while trying to fetch the object.
+            ReadOnlyError: If the provider is in read-only mode.
         """
+        self.check_readonly()
```

Comment:
would be nicer to add a decorator for this, but doesn't really matter

Reply:
I think it's more trouble than it's worth.